### PR TITLE
[BB-248] Improving test coverage

### PR DESCRIPTION
**Pull request by Franr**
## What?
Improving tests coverage, mostly around `modules/credentials`
Separating GHA files into 2: we need the "push" event for the unit tests, for the coverage integration, but not on the integration ones.

## Why?
To improve the tools quality/realiability!

## References
https://coveralls.io/builds/66184725

### CODE REVIEW

File: leverage/modules/credentials.py [hunk 1]
```diff
@@ -565,18 +566,17 @@ def configure_accounts_profiles(profile, region, organization_accounts, project_
         profile(str): Name of the profile to configure.
         region (str): Region.
         organization_accounts (dict): Name and id of all accounts in the organization.
-        project_accounts (dict): Name and email of all accounts in project configuration file.
+        project_accounts (list): Name and email of all accounts in project configuration file.
         fetch_mfa_device (bool): Whether to fetch MFA device for profiles.
     """
-    short_name, type = profile.split("-")
```
Comment (Franr) [inline leverage/modules/credentials.py:571]: `type` is a built-in function, better to not shadow it :slightly_smiling_face:

File: tests/test_modules/test_credentials.py [hunk 2]
```diff
@@ -59,3 +78,191 @@ def test_load_configs_for_credentials(with_click_context):
...
+
+
+@pytest.mark.parametrize("mfa_device", [False, True])
+@mock.patch("leverage.modules.credentials._get_mfa_serial", new=Mock(return_value="mfa123"))
+@mock.patch("leverage.modules.credentials._backup_file")
+def test_configure_accounts_profiles(mocked_backup, mfa_device, muted_click_context):
+    """
+    Test that the expected jsons for the aws credentials are generated as expected.
+    Cover both mfa and no-mfa cases.
+    """
```
Comment (borland667) [inline tests/test_modules/test_credentials.py:90]: @Franr I'd split these tests and have a separate test for each case.

**Comment by borland667** [comment]
@Franr General question. Perhaps I'm not familiar with the code yet but I'm finding it hard to understand what are you trying to test with some of the methods that do not describe what you are trying to test. Can we add docstrings with descriptions to all tests so its clear what is the intent ?

File: tests/test_modules/test_credentials.py [hunk 3]
```diff
@@ -59,3 +78,191 @@ def test_load_configs_for_credentials(with_click_context):
...
+
+
+def test_backup_file():
+    with mock.patch("leverage.modules.credentials.AWSCLI", mocked_aws_cli):
+        _backup_file("config")
+
+    assert (
+        mocked_aws_cli.system_exec.call_args_list[0][0][0] == "sh -c 'cp $AWS_CONFIG_FILE \"${AWS_CONFIG_FILE}.bkp\"'"
+    )
+
```
Comment (borland667) [inline tests/test_modules/test_credentials.py:200]: this one is testing implementation details of the external library

**Review by borland667 — APPROVED** [review]

**Review by juanmatias — APPROVED** [review]

**Pull request merged**